packages/flutter/lib/src/painting/_network_image_io.dart

```diff
@@ -57,63 +58,79 @@ class NetworkImage extends image_provider.ImageProvider<image_provider.NetworkIm
     );
   }
 
-  // Do not access this field directly; use [_httpClient] instead.
-  // We set `autoUncompress` to false to ensure that we can trust the value of
-  // the `Content-Length` HTTP header. We automatically uncompress the content
-  // in our call to [consolidateHttpClientResponseBytes].
-  static final HttpClient _sharedHttpClient = HttpClient()..autoUncompress = false;
+  static Future<Isolate> _pendingLoader;
+  static List<Function> _pendingLoaderUsers;
+  static SendPort _requestPort;
 
-  static HttpClient get _httpClient {
-    HttpClient client = _sharedHttpClient;
-    assert(() {
-      if (debugNetworkImageHttpClientProvider != null)
-        client = debugNetworkImageHttpClientProvider();
-      return true;
-    }());
-    return client;
-  }
-
-  Future<ui.Codec> _loadAsync(
+  Future<ui.Codec> _loadAsyncOnDesignatedIsolate(
     NetworkImage key,
     StreamController<ImageChunkEvent> chunkEvents,
   ) async {
     try {
       assert(key == this);
 
       final Uri resolved = Uri.base.resolve(key.url);
-      final HttpClientRequest request = await _httpClient.getUrl(resolved);
-      headers?.forEach((String name, String value) {
-        request.headers.add(name, value);
+
+      final Completer<TransferableTypedData> completer = Completer<TransferableTypedData>();
+      final RawReceivePort port = RawReceivePort((dynamic message) {
```

Comment:
Side-note: can we beef up the docs on `RawReceivePort`?  There are no class-level docs, and there's no docs on what the signature of `handler` is supposed to be in the constructor.

Reply:
Please do follow-up on this.  We should be improving docs at the point where we see that they're lacking.

Reply:
> no class-level docs

Are you talking about https://api.dartlang.org/stable/2.4.0/dart-isolate/RawReceivePort/RawReceivePort.html ?

Reply:
https://api.dartlang.org/stable/2.4.0/dart-isolate/RawReceivePort-class.html

Reply:
I meant if you click on [...] at end of `Opens a long-lived port for receiving messages. [...]` you will another paragraph with few more details about the class.

Reply:
> what's the difference between a RawReceivePort and a ReceivePort?

Conceptualy ReceivePort extends Stream class, so provides all belts and whistles that come with it. RawReceivePort on the other hand is just a message handler.

> We're using RawReceivePort.sendPort here -- is that the only way to get an instance of SendPort? 

Yeah, the only way to get SendPort is to have ReceivePort first, then ask it for it's SendPort: https://api.dartlang.org/stable/2.0.0/dart-isolate/SendPort/send.html

cc @lrhn

Reply:
There's also no common interface between `ReceivePort` and `RawReceivePort`, making them again seem like two entirely different things.

tldr: very explicit documentation would help 🙂

Reply:
Another case that docs would help for: I didn't see anywhere that the existence of a `RawReceivePort` would keep an isolate alive (@aam just told me).